platform/osgi/platform-osgi-condpermadmin/src/main/java/org/codice/ddf/condpermadmin/PermissionActivator.java

```diff
@@ -0,0 +1,205 @@
+import net.sourceforge.prograde.policyparser.ParsedPolicyEntry;
+import net.sourceforge.prograde.policyparser.ParsedPrincipal;
+import net.sourceforge.prograde.policyparser.Parser;
+import net.sourceforge.prograde.type.Priority;
+import org.apache.commons.lang3.text.StrSubstitutor;
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.condpermadmin.ConditionInfo;
+import org.osgi.service.condpermadmin.ConditionalPermissionAdmin;
+import org.osgi.service.condpermadmin.ConditionalPermissionInfo;
+import org.osgi.service.condpermadmin.ConditionalPermissionUpdate;
+import org.osgi.service.permissionadmin.PermissionInfo;
+import org.osgi.util.tracker.ServiceTracker;
+
+/** Initializes the CondPermAdmin permission table from all policy files in ddf_home/security */
+public class PermissionActivator implements BundleActivator {
+
+  private static final String BUNDLE_NAME_CONDITION =
+      "org.codice.ddf.condition.BundleNameCondition";
+
+  private static final String PRINCIPAL_CONDITION = "org.codice.ddf.condition.PrincipalCondition";
+
+  private static final String SIGNER_CONDITION = "org.codice.ddf.condition.SignerCondition";
+
+  private static final String OSGI_CODEBASE = "file:/";
+
+  private static final String PERM_ADMIN_SERVICE_NAME =
+      "org.osgi.service.condpermadmin.ConditionalPermissionAdmin";
+
+  private ServiceTracker<ConditionalPermissionAdmin, ConditionalPermissionAdmin> permAdminTracker;
+
+  @Override
+  public void start(BundleContext bundleContext) throws Exception {
+    System.setProperty("/", File.separator);
+    permAdminTracker = new ServiceTracker<>(bundleContext, PERM_ADMIN_SERVICE_NAME, null);
+    permAdminTracker.open();
+
+    ConditionalPermissionAdmin conditionalPermissionAdmin = permAdminTracker.getService();
+    String policyDir = SecurityActions.getSystemProperty("ddf.home") + File.separator + "security";
+    if (policyDir.startsWith("=")) {
+      policyDir = policyDir.substring(1);
+    }
+    File policyDirFile = new File(policyDir);
+    List<ParsedPolicy> parsedPolicies = new ArrayList<>();
+    for (File file : Objects.requireNonNull(policyDirFile.listFiles())) {
+      parsedPolicies.add(new Parser(false).parse(new FileReader(new File(file.getAbsolutePath()))));
+    }
+    ConditionalPermissionUpdate conditionalPermissionUpdate =
+        conditionalPermissionAdmin.newConditionalPermissionUpdate();
+    Priority priorityResult = null;
+    List<ConditionalPermissionInfo> allGrantInfos = new ArrayList<>();
+    List<ConditionalPermissionInfo> allDenyInfos = new ArrayList<>();
+    for (ParsedPolicy parsedPolicy : parsedPolicies) {
+      List<ParsedPolicyEntry> grantEntries = parsedPolicy.getGrantEntries();
+      List<ParsedPolicyEntry> denyEntries = parsedPolicy.getDenyEntries();
+      List<ConditionalPermissionInfo> grantInfos = new ArrayList<>();
+      List<ConditionalPermissionInfo> denyInfos = new ArrayList<>();
```

Comment:
❓ Why create temporary lists just to add all entries computed in them back to the `allXXXInfos` list after? Why not just pass the `allXXXInfos` list directly?

Reply:
Just made it easier to debug. I can get rid of it now.